pkg/syncer/resourcesync/controller.go

```diff
@@ -312,6 +314,21 @@ func (c *controller) process(ctx context.Context, key string) error {
 		c.notifySubscribers(ctx)
 	}
 
+	if err := c.PatchSyncTargetCondition(ctx, unauthorizedGVRs, syncTarget); err != nil {
+		errs = append(errs, err)
+	}
+
+	return errors.NewAggregate(errs)
+}
+
+type GVRAuthorizationChecker struct {
+	DownstreamKubeClient kubernetes.Interface
+	SyncTargetClient     workloadv1alpha1typed.SyncTargetInterface
+}
+
+func (c *GVRAuthorizationChecker) PatchSyncTargetCondition(ctx context.Context, unauthorizedGVRs []string, syncTarget *workloadv1alpha1.SyncTarget) error {
```

Comment:
why does a checker (whatever that is, but it is hopefully side-effect free) some patching?

Reply:
The whole `GVRAuthorizationChecker` is removed.